packages/wrangler/src/__tests__/user.test.ts

```diff
@@ -50,6 +53,26 @@ describe("wrangler", () => {
         expiration_time: expect.any(String),
       });
     });
+
+    // TODO: Improve OAuth mocking to handle `/token` endpoints from different calls
+    it("should confirm error message and output for failed login", async () => {
+      mockOAuthServerCallback();
+      writeAuthConfigFile({
+        oauth_token: "jklkl",
+        // expiration_time: undefined,
+        refresh_token: "jkl",
+      });
+      mockGrantAuthorization({ respondWith: "success" });
+
+      mockExchangeRefreshTokenForAccessToken({
+        respondWith: "badResponse",
+      });
+
+      await expect(requireAuth({} as Config, false)).rejects.toThrowError();
```

Comment:
Using `requireAuth()` here rather than `wrangler login` is correct since we never try to do a refresh during the normal login flow. It only occurs when we are checking that we are authenticated already.

But perhaps this test should be moved outside of the `wrangler login` describe block?

Reply:
I agree, will do!